WS2012R2/lisa/setupscripts/STOR_VSS_Backup_Change_Hypervvssd.ps1

```diff
@@ -0,0 +1,203 @@
+    </test>
+
+.Parameter vmName
+    Name of the VM to backup
+
+.Parameter hvServer
+    Name of the Hyper-V server hosting the VM.
+
+.Parameter testParams
+    Test data for this test case
+
+.Example
+
+    .\setupscripts\STOR_VSS_Backup_Change_Hypervvssd.ps1 -hvServer localhost -vmName vm_name -testParams 'driveletter=D:;RootDir=path/to/testdir;sshKey=sshKey;ipv4=ipaddress'
+
+#>
+
+param([string] $vmName, [string] $hvServer, [string] $testParams)
+
+#######################################################################
+#
+# Main script body
+#
+#######################################################################
+
+# Check input arguments
+if ($vmName -eq $null)
+{
+    "ERROR: VM name is null"
+    return $False
+}
+
+# Check input params
+$params = $testParams.Split(";")
+
+foreach ($p in $params)
+{
+  $fields = $p.Split("=")
+
+  switch ($fields[0].Trim())
+    {
+    "TC_COVERED" { $TC_COVERED = $fields[1].Trim() }
+    "sshKey" { $sshKey = $fields[1].Trim() }
+    "ipv4" { $ipv4 = $fields[1].Trim() }
+    "rootdir" { $rootDir = $fields[1].Trim() }
+    "driveletter" { $driveletter = $fields[1].Trim() }
+     default  {}
+    }
+}
+
+if ($null -eq $sshKey)
+{
+    "ERROR: Test parameter sshKey was not specified"
+    return $False
+}
+
+if ($null -eq $ipv4)
+{
+    "ERROR: Test parameter ipv4 was not specified"
+    return $False
+}
+
+if ($null -eq $rootdir)
+{
+    "ERROR: Test parameter rootdir was not specified"
+    return $False
+}
+
+if ($null -eq $driveletter)
+{
+    "ERROR: Backup driveletter is not specified."
+    return $False
+}
+
+# Change the working directory to where we need to be
+cd $rootDir
+
+#
+# Delete any summary.log from a previous test run, then create a new file
+#
+$summaryLog = "${vmName}_summary.log"
+del $summaryLog -ErrorAction SilentlyContinue
+Write-output "This script covers test case: ${TC_COVERED}" | Tee-Object -Append -file $summaryLog
+
+# Source TCUtils.ps1 for common functions
+if (Test-Path ".\setupScripts\TCUtils.ps1") {
+	. .\setupScripts\TCUtils.ps1
+	"Info: Sourced TCUtils.ps1"
+}
+else {
+	"Error: Could not find setupScripts\TCUtils.ps1"
+	return $false
+}
+
+# Source STOR_VSS_Utils.ps1 for common VSS functions
+if (Test-Path ".\setupScripts\STOR_VSS_Utils.ps1") {
+	. .\setupScripts\STOR_VSS_Utils.ps1
+	"Info: Sourced STOR_VSS_Utils.ps1"
+}
+else {
+	"Error: Could not find setupScripts\STOR_VSS_Utils.ps1"
+	return $false
+}
+## run set up
+$sts = runSetup $vmName $hvServer $driveletter
+if (-not $sts[-1])
+{
+    return $False
+}
+
+# set the backup type array, if stop hypervvssd, it exectes offline backup, if start hypervvssd, it executes online backup
+$backupTypes = @("offline","online")
+
+# setVSSD uses to set hypervvssd status, firstly stop, then start
+$setVSSD= @($false,$true)
+
+for ($i = 0; $i -le 1; $i++ )
+{
+    if ( $setVSSD[$i] -eq $false)
+    {# stop hypervvssd daemons then do backup
+        $retVal = SendCommandToVM $ipv4 $sshKey "service hypervvssd stop"
```

Comment:
@chvalean Have updated base on your comments, for ubuntu, I'm not sure the exact hyperv daemons name for different release version, so currently skip for it.
Thank you so much.